.github/workflows/build-test.yml

```diff
@@ -49,6 +49,8 @@ jobs:
                 PARAMETERS_SECRET: ${{ secrets.PARAMETERS_SECRET }}
                 CLOUD_PROVIDER: ${{ matrix.cloud }}
               run: ./ci/test.sh
+            - name: Upload coverage to Codecov
```

Comment:
I have a couple of questions :) 
1. How it calculates coverage? Is it based on our `coverage.txt`?
2. I'm wondering how it should work. We have n go versions * m cloud providers * k operating systems. So what goes to codecov? All n * m * k values? What should be shown in the badge?
3. How does the authentication to codecov works? Is it some global snowflake integration?
4. There are a lot of comments of codecov below the PR. Can we limit it only to files changed in this PR?

Reply:
ad 2. in node js we gather coverage only from one job (MacOS+AWS) and publish it to CodeCov
If you want to publish more than you should add some additional tags to codecov to distiniguish between coverages?

Reply:
1. It uses the formula `hits / (sum of hit + partial + miss)` https://docs.codecov.com/docs/about-code-coverage
2. Codecov merges all the files uploaded. So when the tests are run n * m * k files (18 uploads) are uploaded and merged to get the total coverage.
3. I'm not sure of the auth configuration, it was already set up. Maybe @sfc-gh-igarish can answer this one.
4. The codecov run here is comparing to a base commit from 3 years ago when the gosnowflake project was last uploaded there. There was only 23 files then and now there are 48 plus a lot more code in the files. We need to merge this to get a new baseline.

Reply:
1. Ok, I was wondering how it integrates with coverage.txt, but I see that it is automatic.
2. I wonder what is the policy. But looking at codecov I believe that it combines all lines from all builds to check if a line is tested in any build.
3. Ok.
4. Do we need to do it manually? How to keep codecov updated automatically?

Reply:
After the change is merged it will run against master, same as the [PHP PR](https://github.com/snowflakedb/pdo_snowflake/pull/330).